Transformations: Convert field types to time string number or boolean

nikki-kiga opened the pull request:
**What this PR does / why we need it**:
Can help to quickly convert specific fields to a target type. In particular, for time fields that are in a string format.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

For a CSV example: before and after
<img width="983" alt="example-csv-before" src="https://user-images.githubusercontent.com/42276368/130883337-b957ff53-9c6a-41d9-a81d-4441a6963239.png">
<img width="984" alt="example-csv-after" src="https://user-images.githubusercontent.com/42276368/130883329-3ab12a6f-3d01-4a4c-a8c6-e83d940ca68d.png">

nikki-kiga committed:
outline string to time

add stringToTime transformer

start to add format

add type and dateformat

rename stringToTime to fieldConversion

add more type support and use FieldNamePicker

add field conversion transformation

ryantxu commented on packages/grafana-data/src/transformations/transformers/fieldConversion.test.ts:
```diff
@@ -0,0 +1,152 @@
+import { mockTransformationsRegistry } from '../../utils/tests/mockTransformationsRegistry';
+import { ensureTimeField, fieldConversion, fieldConversionTransformer } from './fieldConversion';
+
+const stringTime = toDataFrame({
```
for the tests, it is nicer to see the input and output closer... maybe rather than defined constants at the top, just stick them inline?

leeoniya commented on packages/grafana-data/src/transformations/transformers/fieldConversion.ts:
```diff
@@ -0,0 +1,171 @@
+    if (value) {
+      let parsed;
+      if (dateFormat) {
+        parsed = dateTimeParse(value, { format: dateFormat });
```
probably this?

```suggestion
        parsed = dateTimeParse(value, { format: dateFormat }).valueOf();
```

nikki-kiga committed:
adjust for performance feedback

rename and adjust labels and widths

shorten labels and null values

rename to convertFieldType

update test

torkelo reviewed (commented):
Wonder if this transform should be named something relating to type instead

* Value type converter 
* Convert value types 
* Change field type
* Convert field type

dprokop commented:
This will be super helpful in scenaris like this: https://github.com/grafana/grafana/issues/34693

dprokop reviewed (commented):
Left coupleof comments @nikki-kiga, but all in all this looks good!

nikki-kiga committed:
make updates

ryantxu reviewed (approved):
@nikki-kiga -- this looks great, I think we should merge and follow up with two more PRs:

1. finishing touches
* docs for this transformer
* use name matchers rather than directly comparing the field names

2. improve the FieldNamePicker
* add "details" setting to the options that would include the type and first few values in the description
